OneCommunityGlobal/HGNRest
# Ilya consumables add page

# Description
<img width="754" alt="consumables-add" src="https://github.com/OneCommunityGlobal/HGNRest/assets/105448137/b7826a97-cf81-484e-a1d1-4b4f342d8699">

Adds functionality for the user to navigate to /consumables/add, enter the required information (consumable name, description, measurement units, size) and create a new consumable type in the system. 

The front end validates the input to have a name that is at least 4 characters long and a description that is at least 10 characters long. The back end checks if a consumable under the same name already exists.

## Related PRS (if any):
This backend PR is related to the [#1926](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1926) front end PR.
To test this backend PR you need to checkout the **ilya_consumables_add_page** branch on the front end 
…

## Main changes explained:
- /consumables GET route added to fetch existing consumable types
- /consumables POST route added to create a new consumable type
- fetchConsumableTypes and addConsumableType functions added to bmInventoryTypeController.js to find and create using the consumableType model
- consumableType model edited to include an optional "size" parameter, "units" is now required

## How to test:
1. check into current branch
2. do `npm install` and `npm run build && npm start` to run this PR locally
3. checkout **ilya_consumables_add_page** on the front end, do `npm install` and `npm run start:local` to run on the front end 
4. log in as any user
5. go to [http://localhost:3000/bmdashboard](http://localhost:3000/bmdashboard) and log in again
6. go to [http://localhost:3000/bmdashboard/consumables/add](http://localhost:3000/bmdashboard/consumables/add), enter some dummy data, press "Add Consumable" when the button is enabled
8. A success message should appear if everything is acceptable or a warning message if not (such as a duplicate value)
<img width="1127" alt="consumables-fe" src="https://github.com/OneCommunityGlobal/HGNRest/assets/105448137/bb2191c9-5b19-4fe1-b2ac-55477721b9dc">
9. In MongoDB compass open hgnData_dev.buildingInventoryTypes collection. Press "Find", scroll all the way to the bottom or search by name and verify that the entered information is saved in the database
<img width="1419" alt="compass" src="https://github.com/OneCommunityGlobal/HGNRest/assets/105448137/a6ac8f3a-e985-47b5-b167-09efbb3276ea">

## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HGNRest/assets/105448137/a3890930-fc88-4915-93fa-c385ce635ef2



## Note:
Examples of construction consumables can include nails, screws, batteries, lumber, etc.
